dropbox/dropbox.py

```diff
@@ -452,6 +466,39 @@ def request_json_string(self,
         else:
             raise HttpError(request_id, r.status_code, r.text)
 
+    def with_path_root(self, path_root=None):
+        """
+        Creates a new object with the same parameters as this object, but also
+        sets the ``Dropbox-API-Path-Root` header to :param:`path_root`.
+
+        :param PathRoot path_root: the path root object to serialize and use for
+        the ``Dropbox-API-Path-Root`` header. (If ``None``, the header is
+        cleared.)
+        :return: a new instance of this class with the ``Dropbox-API-Path-Root``
+        header set.
+        """
+        headers = self._headers
+
+        if headers is not None:
+            headers = dict(headers)  # make a copy that we can manipulate
+            headers.pop(PATH_ROOT_HEADER)
```

Comment:
This will generate a key error if `PATH_ROOT_HEADER` is not already assigned in `self._headers`

Reply:
Good catch! I'm pretty sure I meant `headers.pop(PATH_ROOT_HEADER, None)`, but I'm refactoring per your next comment.